x-pack/plugins/canvas/common/lib/pivot_object_array.ts

```diff
@@ -6,9 +6,15 @@
 
 import { map, zipObject } from 'lodash';
 
-const isString = val => typeof val === 'string';
+const isString = (val: any): boolean => typeof val === 'string';
 
-export function pivotObjectArray(rows, columns) {
+export function pivotObjectArray<
+  RowType extends { [key: string]: any },
+  ReturnColumns extends string | number | symbol = keyof RowType
+>(
+  rows: RowType[],
+  columns?: string[]
+): { [s in ReturnColumns]: s extends keyof RowType ? Array<RowType[s]> : undefined[] } {
```

Comment:
nit: make `s` a whole word variable: why not `column`?  And do we expect an array of `undefined`s?  I think you want `never`.